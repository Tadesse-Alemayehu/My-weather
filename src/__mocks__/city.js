export default {
  id: 'c58ca307-8097-48f0-91f7-5869070696c3',
  name: 'Wereta',
  distance: 43.49,
  population: 26813,
  nearMe: 1,
  country: 'Ethiopia',
  countryCode: 'ET',
  clouds: {
    all: 46,
  },
  coord: {
    lon: 37.7,
    lat: 11.9167,
  },
  weather: [
    {
      id: 802,
      main: 'Clouds',
      description: 'scattered clouds',
      icon: '03n',
    },
  ],
  unit: 'â„ƒ',
  main: {
    temp: 24.77,
    feels_like: 24.29,
    temp_min: 24.77,
    temp_max: 24.77,
    pressure: 1012,
    humidity: 38,
    sea_level: 1012,
    grnd_level: 826,
  },
  wind: {
    speed: 6,
    deg: 324,
    gust: 9.91,
  },
};
