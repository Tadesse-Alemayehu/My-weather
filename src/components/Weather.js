import React from 'react'
import { Card, Col } from 'react-bootstrap';
import { useNavigate } from 'react-router-dom'
export default function Weather({ city }) {
  const navigate = useNavigate();
  return (
    <Col xs={6} sm={4} lg={2} className="p-0" onClick={() => navigate(`/detail?id=${city.id}`)}>
      <Card className="w-100 h-100">
        <Card.Body>
          <Card.Title>
            {city.name}
          </Card.Title>
          <Card.Text>{city.weather[0].description}</Card.Text>
        </Card.Body>
        <small>{city.main.temp}{city.unit}</small>
        <small>{city.country}</small>
      </Card>
    </Col >
  )
}
