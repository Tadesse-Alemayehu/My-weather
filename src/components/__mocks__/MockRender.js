import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/configureStore';

export default (component) => (
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>
);
