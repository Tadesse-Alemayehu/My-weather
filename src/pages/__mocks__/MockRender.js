import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/configureStore';

const MockRender = (component) => (
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>
);

export default MockRender;
