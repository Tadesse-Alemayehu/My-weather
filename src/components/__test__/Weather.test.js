
import renderer from 'react-test-renderer'
import Weather from '../Weather'
import city from '../../__mocks__/city'
import MockRender from '../__mocks__/MockRender'
describe('Snapshot the check for the FilteredCities', () => {

  test('should match the snapshot', () => {
    const weatherSnap = renderer
      .create(
        MockRender(<Weather city={city} />)
      ).toJSON();
    expect(weatherSnap).toMatchSnapshot();
  })
})