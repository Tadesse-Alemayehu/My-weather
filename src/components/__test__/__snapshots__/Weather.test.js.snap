// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot the check for the FilteredCities should match the snapshot 1`] = `
<div
  className="p-0 col-lg-3 col-sm-6 col-6"
  onClick={[Function]}
  title="city-weather"
>
  <div
    className="w-100 h-100 card"
  >
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Wereta
      </div>
      <p
        className="card-text"
      >
        scattered clouds
      </p>
    </div>
    <small>
      24.77
      â„ƒ
    </small>
    <small>
      Ethiopia
    </small>
    <svg
      className="weather-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.5 3a4.002 4.002 0 0 0-3.8 2.745.5.5 0 1 1-.949-.313 5.002 5.002 0 0 1 9.654.595A3 3 0 0 1 13 12H4.5a.5.5 0 0 1 0-1H13a2 2 0 0 0 .001-4h-.026a.5.5 0 0 1-.5-.445A4 4 0 0 0 8.5 3zM0 7.5A.5.5 0 0 1 .5 7h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5zm2 2a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1-.5-.5zm-2 4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1-.5-.5z"
      />
    </svg>
    <svg
      className="see-detail"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
      />
      <path
        d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
      />
    </svg>
  </div>
</div>
`;
